rcl/test/rcl/test_info_by_topic.cpp

```diff
@@ -357,6 +361,8 @@ TEST_F(
     &subscription_options);
   ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
   const std::string fqdn = std::string("/") + this->topic_name;
+  ASSERT_TRUE(wait_for_established_subscription(&publisher, 10, 100));
+  ASSERT_TRUE(wait_for_subscription_to_be_ready(&subscription, &this->context, 10, 100));
```

Comment:
I am not sure this needs to be called here. I believe that we can confirm that graph cache is constructed vi wait_for_established_subscription.

Reply:
This case tests both `rcl_get_publishers_info_by_topic` and `rcl_get_subscriptions_info_by_topic`, so I think we need to add a new wait_helper(such as `wait_for_established_publisher`) which is similar to `wait_for_established_subscription`. (make sure two directions are established.)

NOTE: `wait_for_subscription_to_be_ready` is used for a subscriber to wait to get message data from a publisher.
it means there should be a `rcl_publish` calling before (or at a new thread) and then `rcl_take` to get message data.
So we don't use it here.